src/server/game/Entities/Player/Player.cpp

```diff
@@ -1887,9 +1887,25 @@ void Player::Regenerate(Powers power)
                 }
             }
             break;
-        case POWER_ENERGY:                                  // Regenerate energy (rogue)
-            addvalue += 0.01f * m_regenTimer * sWorld->getRate(RATE_POWER_ENERGY);
+        case POWER_ENERGY: // Regenerate energy (rogue)
+        {
+            float baseRegenRate = 1.0f; // 1 energy per 0.1 seconds
+            float hasteMultiplier = 1.0f; // Start with no haste
+
+            // Apply haste effects from rogue talents
+            if (HasAuraEffect(13750, 0, ObjectGuid())) // Adrenaline Rush
+                hasteMultiplier *= 2.0f; // 100% faster regeneration
+
+            if (HasAuraEffect(58426, 0, ObjectGuid())) // Overkill
+                hasteMultiplier *= 1.3f; // 30% faster regeneration
+
+            if (HasAuraEffect(61329, 0, ObjectGuid())) // Vitality
+                hasteMultiplier *= 1.25f; // 25% faster regeneration
+
+            // Calculate the actual increment based on the haste multiplier
+            addvalue += baseRegenRate * hasteMultiplier * (m_regenTimer / 1000.0f); // Convert m_regenTimer to seconds
```

Comment:
This looks hacky to me.
Most of specified auras have `SPELL_AURA_MOD_POWER_REGEN_PERCENT` aura effect, right?
Can we utilize that data instead of hardcoded values here?
Also I see that a few lines below, we are already making such calculations:
```
if (power != POWER_MANA)
    {
        AuraEffectList const& ModPowerRegenPCTAuras = GetAuraEffectsByType(SPELL_AURA_MOD_POWER_REGEN_PERCENT);
        for (AuraEffectList::const_iterator i = ModPowerRegenPCTAuras.begin(); i != ModPowerRegenPCTAuras.end(); ++i)
            if (Powers((*i)->GetMiscValue()) == power)
                AddPct(addvalue, (*i)->GetAmount());
        // Butchery requires combat for this effect
        if (power != POWER_RUNIC_POWER || IsInCombat())
            addvalue += float(GetTotalAuraModifierByMiscValue(SPELL_AURA_MOD_POWER_REGEN, power) * ((power != POWER_ENERGY) ? m_regenTimerCount : m_regenTimer)) / (5.0f * IN_MILLISECONDS);
    }
```
What is wrong with this code? Will it conflict with suggested implementation?